gosap.go

```diff
@@ -433,3 +433,210 @@ func retrieveDocument[T any](s *Session, endpoint string) (*T, error) {
 
 	return &doc, nil
 }
+
+func (s *Session) GetInventoryCounting(cfg Config, id int) (*InventoryCounting, error) {
+	url := cfg.GetInventoryCountingEndpoint(id)
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, content, err := s.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	var inventoryCounting InventoryCounting
+
+	// Unmarshal the JSON into the wrapper struct
+	if err := json.Unmarshal(content, &inventoryCounting); err != nil {
+		return nil, fmt.Errorf("could not load json response due to %s", err)
+	}
+
+	return &inventoryCounting, nil
+}
+
+func (s *Session) GetInventoryCountings(cfg Config) ([]InventoryCounting, error) {
+	return s.getInventoryCountings(cfg, cfg.GetInventoryCountingsEndpoint())
+}
+
+func (s *Session) getInventoryCountings(cfg Config, endpoint string) ([]InventoryCounting, error) {
+	req, err := http.NewRequest(http.MethodGet, endpoint, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, content, err := s.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	// Unmarshal the JSON into the wrapper struct
+	var response InventoryCountingResponse
+	if err := json.Unmarshal(content, &response); err != nil {
+		return nil, fmt.Errorf("could not load json response due to %s", err)
+	}
+
+	if response.NextLink != nil && *response.NextLink != "" {
+		next, err := s.getInventoryCountings(cfg, cfg.BuildEndpoint(*response.NextLink))
+		if err != nil {
+			return response.Value, err
+		}
+
+		response.Value = append(response.Value, next...)
+	}
+
+	return response.Value, nil
+}
+
+func (s *Session) CreateInventoryCounting(cfg Config, counting InventoryCounting) (bool, error) {
+	payload, err := json.Marshal(counting)
+	if err != nil {
+		return false, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, cfg.CreateInventoryCountingEndpoint(), strings.NewReader(string(payload)))
+	if err != nil {
+		return false, err
+	}
+
+	resp, _, err := s.Do(req)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
```

Comment:
No need to close the body it's already closed in the Do function. (In the future, It might be better to entirely remove the response and return the content and status)